spec/ddtrace/contrib/rails/rails_semantic_logger_auto_injection_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'ddtrace/contrib/rails/rails_helper'
+
+RSpec.describe 'Rails Log Auto Injection' do
+  include Rack::Test::Methods
+  include_context 'Rails test application'
+
+  let(:routes) do
+    {
+      '/semantic_logger' => 'semantic_logger_test#index'
+    }
+  end
+
+  let(:controllers) do
+    [
+      semantic_logger_controller
+    ]
+  end
+
+  let(:semantic_logger_controller) do
+    stub_const('SemanticLoggerTestController', Class.new(ActionController::Base) do
+      def index
+        Rails.logger.info('MINASWAN')
+        render inline: '<html> <head> </head> <body> <div> Hello from index </div> </body> </html>'
+      end
+    end)
+  end
+
+  before do
+    Datadog.configuration[:rails].reset_options!
+    Datadog.configure do |c|
+      c.use :rails, log_injection: true
+    end
+
+    allow(ENV).to receive(:[]).and_call_original
+  end
+
+  after do
+    Datadog.configuration[:rails].reset_options!
+  end
+
+  context 'with Log_Injection Enabled' do
+    # defined in rails support apps
+    let(:logs) { log_output.string }
+
+    if Rails.version >= '4.0'
+      context 'with Semantic Logger' do
+        # for logsog_injection testing
+        require 'rails_semantic_logger'
+        subject(:response) { get '/semantic_logger' }
+
+        before do
+          allow(ENV).to receive(:[]).with('USE_SEMANTIC_LOGGER').and_return(true)
+        end
+
+        after do
+          SemanticLogger.close
+        end
+
+        context 'with semantic logger enabled' do
+          context 'with semantic logger setup and no log_tags' do
+            it 'injects trace_id into logs' do
+              is_expected.to be_ok
+              # force flush
+              SemanticLogger.flush
+
+              expect(logs).to include(spans[0].trace_id.to_s)
```

Comment:
I think it makes sense to cover all fields, as this is brand new code introduced by this PR:
```ruby
              dd: {
                trace_id: correlation.trace_id.to_s,
                span_id: correlation.span_id.to_s,
                env: correlation.env.to_s,
                service: correlation.service.to_s,
                version: correlation.version.to_s
              },
              ddsource: ['ruby']
```